pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/IntRanges.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.core.operator.filter;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.utils.Pairs;
+
+
+/**
+ * Various utilities to deal with integer ranges, with both bounds being inclusive (eg. for docId ranges).
+ */
+public class IntRanges {
```

Comment:
`IntPair` doesn't seem like the right place to add those methods, as the name suggests it's just a pair of integers with no other functionality.

It seems that we only use `IntPair` for `docId` ranges though, so I'll give this another look and see if we can rename this class to `DocIdRange` instead and move those methods there.